internal/characters/xingqiu/burst.go

```diff
@@ -154,6 +182,40 @@ func (c *char) summonSwordWave() {
 	c.AddStatus(burstICDKey, 60, true)
 }
 
+func (c *char) getN0Delay() int {
+	active := c.Core.Player.ActiveChar()
+	activeCharKey := active.Base.Key
+	// The character doesn't have an alt form
+	if XQ_N0_delays_alt_forms[activeCharKey] == -1 {
+		return XQ_N0_delays[activeCharKey]
+	}
+
+	if active.StatusIsActive(alt_form_status_keys[activeCharKey]) {
+		return XQ_N0_delays_alt_forms[activeCharKey]
+	} else {
+		return XQ_N0_delays[activeCharKey]
+	}
+}
+
+func (c *char) burstStateDelayFunc() {
+	//ignore if on ICD
+	if c.StatusIsActive(burstICDKey) {
+		return
+	}
```

Comment:
should add a check for if the character is the same as when it was scheduled. ideally this check should instead be if the current NA is the same as the one that was scheduled, but I can't think of a good way to do that today. Should probably just add a TODO for this and create an issue

Reply:
I added a source frame check, hopefully that ensure it works properly?